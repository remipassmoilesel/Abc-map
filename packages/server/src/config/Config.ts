/**
 * Copyright © 2021 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { Algorithm } from 'jsonwebtoken';

export const LOCAL_ENVIRONMENT = 'local';
export const TEST_ENVIRONMENT = 'test';
export const STAGING_ENVIRONMENT = 'staging';

export interface ConfigInput {
  environmentName: string;
  development?: DevelopmentDataConfig;
  // External URL is the url that will be used by users. It must contain protocol and full domain. E.g: 'https://abc-map.fr'.
  externalUrl: string;
  server: ServerConfig;
  project: ProjectConfig;
  database: DatabaseConfig;
  jwt: JwtConfig;
  authentication: AuthenticationConfig;
  registration: RegistrationConfig;
  smtp: SmtpConfig;
  datastore: DatastoreConfig;
  legalMentions: string;
  webapp?: WebappConfig;
  userDocumentation?: UserDocConfig;
}

/**
 * This config is used in application
 */
export interface Config extends ConfigInput {
  webappPath: string;
  userDocumentationPath: string;
}

export interface DevelopmentDataConfig {
  // If set, data will be generated in order to facilitate development (users, projects, ...)
  generateData?: {
    users: number;
    projectsPerUser: number;
  };
  // If true, mails will not be sent but persisted on disk
  persistEmails?: boolean;
}

export interface ServerConfig {
  host: string;
  port: number;
  log: {
    requests: boolean;
    errors: boolean;
    warnings: boolean;
  };
  globalRateLimit: {
    max: number;
    timeWindow: string;
  };
  authenticationRateLimit: {
    max: number;
    timeWindow: string;
  };
}

export interface ProjectConfig {
  maxPerUser: number;
}

export interface DatabaseConfig {
  url: string;
  databaseName?: string;
  username: string;
  password: string;
}

export interface JwtConfig {
  algorithm: Algorithm;
}

export interface AuthenticationConfig {
  secret: string;
  tokenExpiresIn: string;
  passwordLostExpiresIn: string;
}

export interface RegistrationConfig {
  passwordSalt: string;
  secret: string;
  confirmationExpiresIn: string;
}

export interface SmtpConfig {
  from: string;
  host: string;
  port: number;
  auth?: {
    user: string;
    pass: string;
  };
}

export interface DatastoreConfig {
  path: string;
}

export interface WebappConfig {
  appendToBody?: string;
}

export interface UserDocConfig {
  appendToBody?: string;
}
