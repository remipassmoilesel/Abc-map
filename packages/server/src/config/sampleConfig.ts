/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 *
 *
 *
 */

import { ConfigInput } from './Config';

/**
 * This function return a valid javascript configuration example.
 */
export function sampleConfig(): string {
  const config: ConfigInput = {
    environmentName: 'local',
    externalUrl: 'http://localhost:10082/',
    server: {
      host: 'localhost',
      port: 10_082,
      log: {
        requests: true,
        errors: true,
        warnings: false,
      },
      globalRateLimit: {
        max: 1000,
        timeWindow: '1m',
      },
      authenticationRateLimit: {
        max: 1000,
        timeWindow: '1m',
      },
    },
    project: {
      maxPerUser: 10,
    },
    database: {
      url: 'mongodb://localhost:27019',
      username: 'admin',
      password: 'admin',
    },
    jwt: {
      algorithm: 'HS512',
    },
    authentication: {
      secret: 'azerty1234',
      tokenExpiresIn: '6h',
      passwordLostExpiresIn: '30min',
    },
    registration: {
      passwordSalt: 'azerty1234',
      secret: 'azerty1234',
      confirmationExpiresIn: '24h',
    },
    smtp: {
      from: 'no-reply@abc-map.fr',
      host: 'smtp.ethereal.email',
      port: 587,
      auth: {
        user: 'lelia16@ethereal.email',
        pass: '63rntn3G4DU3uue2MJ',
      },
    },
    datastore: {
      path: 'resources/dev-datastore',
    },
    development: {
      generateData: {
        users: 100,
        projectsPerUser: 5,
      },
      persistEmails: true,
    },
    webapp: {
      appendToBody: '<div>Server templated content (webapp)</div>',
    },
    userDocumentation: {
      appendToBody: '<div>Server templated content (userDocumentation)</div>',
    },
    legalMentions: `<div>This platform is offered in order to be useful, but ....</div>`,
  };

  return `
module.exports = ${JSON.stringify(config, null, 2)}
`;
}
