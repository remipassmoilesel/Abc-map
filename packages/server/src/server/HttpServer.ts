/**
 * Copyright Â© 2021 RÃ©mi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { Language, Logger } from '@abc-map/shared';
import { Config } from '../config/Config';
import { Services } from '../services/services';
import * as path from 'path';
import { fastify, FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
import fastifySensible from '@fastify/sensible';
import { Controller } from './Controller';
import { jwtPlugin } from './helpers/jwtPlugin';
import fastifyRateLimit from '@fastify/rate-limit';
import helmet from '@fastify/helmet';
import fastifyView from '@fastify/view';
import { getLang } from './helpers/getLang';
import { HookHandlerDoneFunction } from 'fastify/types/hooks';
import { FastifyError } from '@fastify/error';
import { privateControllers, publicControllers } from './controllers';
import metricsPlugin from 'fastify-metrics';
import { defaultRateLimitConfig } from './helpers/defaultRateLimitConfig';
import { hashRequestSource } from './helpers/hashRequestSource';
import { isItWorthLogging } from './helpers/isItWorthLogging';

const logger = Logger.get('HttpServer.ts', 'info');

/**
 * API and webapp server.
 *
 * Here responses are not compressed, this should be handled in reverse proxy (Envoy per example)
 *
 */
export class HttpServer {
  public static create(config: Config, services: Services): HttpServer {
    const controllers = {
      public: publicControllers(config, services),
      private: privateControllers(config, services),
    };
    return new HttpServer(config, services, controllers);
  }

  private app: FastifyInstance;

  private debugRoutes = false;
  private routes: string[] = [];

  constructor(private config: Config, private services: Services, private controllers: { public: Controller[]; private: Controller[] }) {
    // This server is designed to be used behind a TLS proxy
    this.app = fastify({ trustProxy: true });
  }

  /**
   * Start HTTP server.
   *
   * Returned promise resolve when server is closed.
   */
  public async listen(): Promise<void> {
    const { host, port } = this.config.server;

    return new Promise((resolve, reject) => {
      this.app.addHook('onClose', () => resolve());

      this.app
        .listen({ port, host })
        .then(() => logger.info(`Abc-Map server listening on http://${host}:${port}`))
        .catch((err) => reject(err));
    });
  }

  public getApp(): FastifyInstance {
    return this.app;
  }

  public shutdown() {
    return this.app.close();
  }

  public async initialize(): Promise<void> {
    if (this.debugRoutes) {
      this.app.addHook('onRoute', (routeOptions) => {
        this.routes.push(
          JSON.stringify({
            method: routeOptions.method,
            url: routeOptions.url,
            prefix: routeOptions.prefix,
          })
        );
      });
    }

    if (this.config.server.log.requests) {
      this.app.addHook('onRequest', this.logRequest);
    }

    // Global error handler
    this.app.setErrorHandler(this.handleError);

    // Rate limit
    await this.app.register(fastifyRateLimit, {
      global: true,
      ...defaultRateLimitConfig(this.config),
    });

    // Utility methods
    await this.app.register(fastifySensible);

    // Metrics
    await this.app.register(metricsPlugin, { defaultMetrics: { enabled: false }, clearRegisterOnInit: true });

    // Templating engine
    await this.app.register(fastifyView, {
      engine: { handlebars: require('handlebars') },
      root: path.resolve(__dirname, '../../public'),
      viewExt: 'html',
    });

    // Add security headers
    // We allow iframes for shared maps
    await this.app.register(helmet, { contentSecurityPolicy: false, frameguard: false });

    // Public controllers
    await this.app.register(async (app) => {
      this.controllers.public.forEach((ctr) => app.register(ctr.setup, { prefix: ctr.getRoot() }));
    });

    // Private controllers, authentication needed
    await this.app.register(async (app) => {
      this.authenticationHook(app);
      this.controllers.private.forEach((ctr) => app.register(ctr.setup, { prefix: ctr.getRoot() }));
    });

    if (this.debugRoutes) {
      logger.info('Registered routes: ');
      this.routes.forEach((route) => logger.info(route));
    }
  }

  private handleError = (err: FastifyError, request: FastifyRequest, reply: FastifyReply) => {
    const { metrics } = this.services;

    // Log if needed
    if (this.config.server.log.errors) {
      const logTrace = {
        date: new Date().toISOString(),
        error: err.message,
        source: hashRequestSource(request, !this.config.server.log.warnings),
        userAgent: request.headers['user-agent'],
        method: request.method,
        url: request.url,
        status: reply.statusCode,
        stack: err.stack,
      };

      logger.log(JSON.stringify(logTrace));
    }

    // Return 429 error page if necessary
    if (reply.statusCode === 429) {
      void reply.view('webapp/error429', error409Parameters(getLang(request)));
      metrics.requestQuotaExceeded();
      return;
    }
    // Or return error and error status
    else {
      void reply.status(reply.statusCode || 500).send(err);
    }
  };

  private logRequest = (request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction) => {
    if (!isItWorthLogging(request)) {
      done();
      return;
    }

    const logTrace = {
      date: new Date().toISOString(),
      source: hashRequestSource(request, !this.config.server.log.warnings),
      userAgent: request.headers['user-agent'],
      method: request.method,
      url: request.url,
      status: reply.statusCode,
    };

    logger.log(JSON.stringify(logTrace));
    done();
  };

  private authenticationHook(app: FastifyInstance) {
    jwtPlugin(this.config, app);
    app.addHook('onRequest', async (request, reply) => {
      try {
        await request.jwtVerify();
      } catch (err) {
        reply.forbidden();
      }
    });
  }
}

interface Error409Parameters {
  lang: string;
  title: string;
  message: string;
}

export function error409Parameters(lang: Language): Error409Parameters {
  switch (lang) {
    case Language.French:
      return {
        lang,
        title: 'Quota de demandes dÃ©passÃ© ðŸ˜­',
        message: 'Nous avons reÃ§u trop de demandes en provenance de votre adresse IP, veuillez rÃ©essayer plus tard.',
      };
    case Language.English:
      return {
        lang,
        title: 'Quota of requests exceeded ðŸ˜­',
        message: 'We have received too many requests from your IP address, please try again later.',
      };
  }
}
