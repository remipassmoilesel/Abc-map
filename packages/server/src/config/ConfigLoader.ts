/**
 * Copyright © 2021 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { Env, EnvKey } from './Env';
import { Config, ConfigInput } from './Config';
import { errorMessage, Logger } from '@abc-map/shared';
import * as path from 'path';
import * as _ from 'lodash';
import { Validation } from '../utils/Validation';
import { isDirectory } from '../utils/isDirectory';
import { sampleConfig } from './sampleConfig';

export const logger = Logger.get('ConfigLoader.ts', 'info');

export type LoadConfigFunc = () => Promise<Config>;

export class ConfigLoader {
  public static readonly DEFAULT_CONFIG = 'resources/configuration/development.js';

  public static getPathFromEnv(): string {
    const env = new Env();
    return env.get(EnvKey.CONFIG) || ConfigLoader.DEFAULT_CONFIG;
  }

  public static async load(): Promise<Config> {
    const configPath = ConfigLoader.getPathFromEnv();
    return new ConfigLoader().load(configPath);
  }

  public static safeConfig(config: Config): Config {
    const safe: Config = _.cloneDeep(config);
    const safeValue = '<value-masked>';
    safe.authentication.secret = safeValue;
    safe.database.password = safeValue;
    safe.registration.passwordSalt = safeValue;
    safe.registration.secret = safeValue;
    if (safe.smtp.auth?.pass) {
      safe.smtp.auth.pass = safeValue;
    }

    safe.legalMentions = `${safe.legalMentions.substring(0, 50)}...`;

    return safe;
  }

  public async load(configPath: string): Promise<Config> {
    const _configPath = path.resolve(configPath);

    const explainedError = (message: string) => {
      logger.error(
        [
          '',
          '',
          '⚠️   Invalid configuration. ',
          '',
          'Message: ' + message,
          '',
          'This is a valid configuration example:',
          '',
          '------ config.js ------',
          sampleConfig()
            .split('\n')
            .map((line) => '   ' + line)
            .join('\n'),
          '------ config.js ------',
          '',
          'You can see other examples here: https://gitlab.com/abc-map/abc-map/-/tree/master/packages/server/resources/configuration',
          '',
          '',
        ].join('\n')
      );
    };

    let input: ConfigInput;
    try {
      /* eslint-disable @typescript-eslint/no-var-requires */
      input = _.cloneDeep(require(_configPath));
    } catch (err) {
      explainedError(errorMessage(err));
      return Promise.reject(new Error(`Cannot load configuration ${configPath}: ${errorMessage(err)}`));
    }

    // We validate config input
    if (!Validation.ConfigInput(input)) {
      explainedError(Validation.formatErrors(Validation.ConfigInput));
      return Promise.reject(new Error(`Configuration ${configPath} is not valid: ${Validation.formatErrors(Validation.ConfigInput)}`));
    }

    const config: Config = {
      ...input,
      webappPath: path.resolve(__dirname, '../../public/webapp'),
      userDocumentationPath: path.resolve(__dirname, '../../public/user-documentation'),
    };

    // We check if datastore is a directory
    if (!(await isDirectory(config.datastore.path))) {
      throw new Error(`Datastore root '${config.datastore.path}' must be a directory`);
    }

    // We remove trailing slash if present
    config.externalUrl = config.externalUrl.trim();
    if (config.externalUrl.slice(-1) === '/') {
      config.externalUrl = config.externalUrl.slice(0, -1);
    }

    // We check if webapp path is a directory
    if (!(await isDirectory(config.webappPath))) {
      throw new Error(`Webapp root '${config.webappPath}' must be a directory. Try "abc build" on a developer workstation.`);
    }

    return config;
  }
}
