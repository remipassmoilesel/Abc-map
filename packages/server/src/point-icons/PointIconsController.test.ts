/**
 * Copyright © 2023 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { HttpServer } from '../server/HttpServer';
import { Services, servicesFactory } from '../services/services';
import { ConfigLoader } from '../config/ConfigLoader';
import { assert, expect } from 'chai';
import { Config } from '../config/Config';
import { TestAuthentication } from '../utils/TestAuthentication';
import { GetPointIconsResponse } from '@abc-map/shared';

describe('PointIconsController', () => {
  let config: Config;
  let services: Services;
  let testAuth: TestAuthentication;
  let server: HttpServer;

  before(async () => {
    config = await ConfigLoader.load();
    config.server.log.requests = false;
    config.server.log.errors = false;
    config.server.log.warnings = false;

    services = await servicesFactory(config);

    server = HttpServer.create(config, services);
    await server.initialize();

    testAuth = new TestAuthentication(services);
  });

  after(async () => {
    await services.shutdown();
    await server.shutdown();
  });

  describe('GET /api/point-icons', () => {
    it('should fail for non connected user', async () => {
      // Act
      const res = await server.getApp().inject({
        url: '/api/point-icons',
        method: 'POST',
      });

      // Assert
      assert.equal(res.statusCode, 403);
    });

    it('should work for anonymous user', async () => {
      // Act
      const res = await server.getApp().inject({
        url: '/api/point-icons',
        method: 'POST',
        body: {
          names: [
            'iso7001/ISO-7001---Red-Cross.svg',
            'iso7001/ISO-7001---Red-Slash.svg',
            'iso7001/ISO-7001-PI-BP-001.svg',
            'iso7001/ISO-7001-PI-BP-002.svg',
            'iso7001/ISO-7001-PI-BP-003.svg',
          ],
        },
        headers: {
          ...testAuth.anonymous(),
        },
      });

      // Assert
      expect(res.statusCode).equal(200);

      const body: GetPointIconsResponse = JSON.parse(res.body);
      expect(body.icons.length).equal(5);

      expect(body.icons.sort((a, b) => a.name.localeCompare(b.name))).toMatchSnapshot();
    });
  });
});
