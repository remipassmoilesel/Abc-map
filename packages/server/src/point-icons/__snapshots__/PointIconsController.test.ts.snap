// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointIconsController GET /api/point-icons should work for anonymous user 1`] = `
Array [
  Object {
    "content": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"yes\\"?>
<svg version=\\"1.1\\" id=\\"svg5106\\" xmlns:svg=\\"http://www.w3.org/2000/svg\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\" width=\\"150px\\" height=\\"150px\\" viewBox=\\"0 0 150 150\\" enable-background=\\"new 0 0 150 150\\" xml:space=\\"preserve\\">
  <g id=\\"layer1\\" transform=\\"translate(-74.032814,-172.11508)\\">
    <rect id=\\"rect70551\\" x=\\"74.347\\" y=\\"172.115\\" fill=\\"#FFFFFF\\" width=\\"150\\" height=\\"150.001\\"/>
    <path id=\\"rect4508\\" fill=\\"#B71F2E\\" d=\\"M118.923,210.003l-6.738,6.75l30.415,30.36l-30.415,30.362l6.738,6.752&#xD;&#xA;&#x9;&#x9;l30.425-30.375l30.427,30.375l6.736-6.752L156.1,247.116l30.411-30.362l-6.736-6.751l-30.427,30.374L118.923,210.003z\\"/>
  </g>
</svg>",
    "name": "iso7001/ISO-7001---Red-Cross.svg",
  },
  Object {
    "content": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"yes\\"?>
<svg version=\\"1.1\\" id=\\"svg9552\\" xmlns:svg=\\"http://www.w3.org/2000/svg\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" x=\\"0px\\" y=\\"0px\\" width=\\"150px\\" height=\\"150px\\" viewBox=\\"0 0 150 150\\" enable-background=\\"new 0 0 150 150\\" xml:space=\\"preserve\\">
  <g id=\\"layer1\\" transform=\\"translate(-64.052513,-153.87521)\\">
    <rect id=\\"rect70551\\" x=\\"64.053\\" y=\\"153.875\\" fill=\\"#FFFFFF\\" width=\\"150\\" height=\\"150\\"/>
    <g id=\\"g9535\\" transform=\\"translate(71.069615,2.290358)\\">
      <rect id=\\"rect9462\\" x=\\"-8.108\\" y=\\"222.807\\" transform=\\"matrix(-0.7081 -0.7061 0.7061 -0.7081 -43.8683 435.0366)\\" fill=\\"#B71F2E\\" width=\\"152.183\\" height=\\"7.556\\"/>
    </g>
  </g>
</svg>",
    "name": "iso7001/ISO-7001---Red-Slash.svg",
  },
  Object {
    "content": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"yes\\"?>
<svg width=\\"52.917mm\\" height=\\"52.917mm\\" viewBox=\\"0 0 52.917 52.917\\" version=\\"1.1\\" id=\\"svg9008\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:svg=\\"http://www.w3.org/2000/svg\\">
  <defs id=\\"defs9005\\"/>
  <g id=\\"layer1\\" transform=\\"translate(-78.506737,-121.86942)\\">
    <rect style=\\"display:inline;fill:#ffffff;fill-opacity:1;stroke:none;stroke-width:2.01819\\" id=\\"rect70551-3\\" width=\\"52.917\\" height=\\"52.917\\" x=\\"78.506737\\" y=\\"121.86942\\"/>
    <path id=\\"path8966\\" style=\\"display:inline;fill:#000000;stroke-width:0.499999\\" d=\\"m 88.790255,131.46461 v 33.68374 H 99.11469 v -7.44244 h 4.19664 c 0.90603,0.002 1.93322,-0.61396 1.93322,-1.53118 v -5.30872 c 1.18049,-0.79918 1.48448,-1.53102 0.28008,-2.45204 l -0.002,-0.52607 c 1.5801,-0.48943 0.85917,-1.48016 -0.0238,-2.36626 v -2.78588 h 3.59203 c 1.34029,-0.0344 1.60782,-1.52611 0.93793,-2.57504 l -4.50412,-7.53442 v -1.16169 z m 20.166725,13.76867 c -0.82221,0 -1.48414,0.66194 -1.48414,1.48415 v 10.41073 l -2.3611,4.03645 1.18701,4.02663 h 14.84147 v -8.08633 c -0.0103,-0.85281 -0.56939,-1.48164 -1.23455,-1.64021 -0.69464,-0.1656 -1.50353,0.17892 -1.85673,1.18442 l -0.31316,0.0145 c 0.19776,-1.08565 -0.0191,-1.89716 -1.07384,-2.23655 -0.83004,-0.26708 -1.81692,0.0233 -2.07894,1.26297 l -0.60772,-0.13074 c 0.17786,-0.97714 0.17813,-1.5768 -0.77049,-2.096 -0.79933,-0.34588 -2.10088,0.15949 -2.22364,1.54254 l -0.45424,-0.0186 v -8.26978 c 0,-0.82221 -0.66193,-1.48415 -1.48414,-1.48415 z\\"/>
  </g>
</svg>",
    "name": "iso7001/ISO-7001-PI-BP-001.svg",
  },
  Object {
    "content": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"yes\\"?>
<svg width=\\"52.917mm\\" height=\\"52.917mm\\" viewBox=\\"0 0 52.917 52.917\\" version=\\"1.1\\" id=\\"svg1722\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:svg=\\"http://www.w3.org/2000/svg\\">
  <defs id=\\"defs1719\\"/>
  <g id=\\"layer1\\" transform=\\"translate(-105.00616,-66.805695)\\">
    <rect style=\\"fill:#ffffff;fill-opacity:1;stroke:none;stroke-width:2.01819\\" id=\\"rect70551\\" width=\\"52.917\\" height=\\"52.917\\" x=\\"105.00616\\" y=\\"66.805695\\"/>
    <path id=\\"rect74478\\" style=\\"stroke-width:1.01549\\" d=\\"m 127.25872,73.746267 c -1.53718,-1.3e-5 -2.78331,1.246119 -2.7833,2.783293 -1e-5,1.537173 1.24613,2.783305 2.7833,2.783292 1.53717,1.3e-5 2.7833,-1.246119 2.78329,-2.783292 1e-5,-1.537174 -1.24611,-2.783306 -2.78329,-2.783293 z m -3.24011,6.299874 c -0.94725,0.123272 -1.15137,0.687393 -1.40922,1.315682 l -3.40393,10.7797 c -0.20601,0.65242 0.0968,1.32564 0.67903,1.50947 0.58222,0.18384 1.21665,-0.19352 1.42265,-0.84594 l 2.81533,-8.915736 v 6.196004 l -2.66288,18.259869 c -0.12428,0.85226 0.46187,1.63839 1.31413,1.76268 l 0.0372,0.005 c 0.85227,0.12429 1.63839,-0.46134 1.76268,-1.31362 l 2.03089,-13.924727 c 0.0126,-0.44719 1.29055,-0.46481 1.3224,0.0124 l 2.0283,13.912327 c 0.12429,0.85227 0.9104,1.43791 1.76267,1.31362 l 0.0372,-0.005 c 0.85228,-0.12428 1.43842,-0.91042 1.31413,-1.76268 l -2.66287,-18.259869 v -6.196004 l 2.81533,8.915736 c 0.20599,0.65243 0.84044,1.02977 1.42265,0.84594 0.58224,-0.18383 0.88503,-0.85705 0.67903,-1.50947 l -3.40393,-10.7797 c -0.25425,-0.725616 -0.76103,-1.29968 -1.40921,-1.315682 h -1.2175 -1.34979 -2.70681 z m -9.65213,12.214762 a 1,1 0 0 0 -1.00045,0.99993 v 19.521287 h 36.19726 V 93.260833 a 1,1 0 0 0 -0.99993,-0.99993 1,1 0 0 0 -0.99994,0.99993 v 17.521417 h -32.197 V 93.260833 a 1,1 0 0 0 -0.99994,-0.99993 z\\"/>
  </g>
</svg>",
    "name": "iso7001/ISO-7001-PI-BP-002.svg",
  },
  Object {
    "content": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"yes\\"?>
<svg width=\\"52.917mm\\" height=\\"52.917mm\\" viewBox=\\"0 0 52.917 52.917\\" version=\\"1.1\\" id=\\"svg4762\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:svg=\\"http://www.w3.org/2000/svg\\">
  <defs id=\\"defs4759\\"/>
  <g id=\\"layer1\\" transform=\\"translate(-82.63652,-98.811485)\\">
    <rect style=\\"fill:#ffffff;fill-opacity:1;stroke:none;stroke-width:2.01819\\" id=\\"rect70551\\" width=\\"52.917\\" height=\\"52.917\\" x=\\"82.63652\\" y=\\"98.811485\\"/>
    <path id=\\"rect4471\\" style=\\"stroke-width:2;stroke-linecap:round\\" d=\\"m 107.96472,108.56068 c -0.18623,0.004 -0.37042,0.0325 -0.54829,0.0837 -0.51744,0.14907 -0.97923,0.49385 -1.264,0.99632 -0.56954,1.00494 -0.22875,2.28799 0.76378,2.87889 l 0.30695,-0.51573 c -0.7178,-0.42735 -0.95915,-1.34183 -0.54725,-2.06861 0.41189,-0.72679 1.32129,-0.98794 2.05672,-0.59169 0.73543,0.39626 1.01503,1.29903 0.63459,2.04277 l 0.53537,0.27388 c 0.52604,-1.02838 0.13013,-2.29791 -0.88677,-2.84582 -0.33367,-0.17979 -0.69556,-0.26128 -1.0511,-0.25373 z m 1.36684,3.48402 c -0.20311,-5.2e-4 -0.40489,0.0283 -0.59893,0.0842 -0.51745,0.14907 -0.97872,0.49127 -1.26349,0.99373 -0.56955,1.00494 -0.23081,2.29006 0.76171,2.88097 l 0.30851,-0.51573 c -0.71781,-0.42735 -0.9607,-1.34131 -0.5488,-2.0681 0.4119,-0.72677 1.32129,-0.99001 2.05672,-0.59376 0.73543,0.39626 1.01502,1.29904 0.63459,2.04277 l 0.53537,0.27389 c 0.52604,-1.02838 0.13013,-2.29585 -0.88677,-2.84376 -0.31778,-0.17122 -0.66038,-0.25337 -0.99891,-0.25425 z m -2.88199,0.78549 c -0.74028,0 -1.4733,0.368 -1.75183,1.16168 v 0.002 l -2.52181,7.45329 -0.002,0.006 c -0.13517,0.45356 -0.15813,0.92646 0.0413,1.32602 0.19947,0.39956 0.6448,0.66604 1.23248,0.67593 l 0.01,-0.59997 c -0.43388,-0.007 -0.60399,-0.14157 -0.70486,-0.34364 -0.10088,-0.20207 -0.10833,-0.53009 -0.002,-0.88677 l 2.51354,-7.43366 0.002,-0.002 v -0.002 c 0.19564,-0.55168 0.62177,-0.7581 1.18339,-0.7581 z m 4.12688,2.66598 a 1.9620063,1.9620063 0 0 0 -1.96164,1.96164 1.9620063,1.9620063 0 0 0 1.96164,1.96215 1.9620063,1.9620063 0 0 0 1.96215,-1.96215 1.9620063,1.9620063 0 0 0 -1.96215,-1.96164 z m -3.08767,0.85525 c -0.74028,0 -1.47175,0.37059 -1.75028,1.16427 l -0.002,0.002 -2.51922,7.45277 -0.002,0.006 c -0.13518,0.45357 -0.15862,0.92439 0.0408,1.32395 0.19947,0.39956 0.6448,0.6681 1.23249,0.678 l 0.01,-0.59945 c -0.43387,-0.007 -0.60399,-0.14365 -0.70487,-0.34572 -0.10087,-0.20206 -0.11034,-0.53008 -0.004,-0.88676 l 2.5156,-7.43572 v 0.004 c 0.19487,-0.55528 0.62067,-0.76378 1.18391,-0.76378 z m 0.52917,3.47627 c -0.34905,0.003 -0.55053,0.22522 -0.66714,0.52193 l -2.71405,8.59845 c -0.11105,0.35181 0.0828,0.72455 0.4346,0.8356 l 0.0326,0.0103 c 0.35182,0.11105 0.7628,-0.0828 0.87385,-0.4346 l 2.09083,-6.62389 -0.0382,7.94112 -0.26872,10.35648 c -0.0136,0.52414 0.42241,0.94671 0.94672,0.94671 0.52428,0 0.93279,-0.42257 0.94619,-0.94671 0.10534,3.42138 0.24721,-9.95669 0.26872,-10.26553 -0.0445,-0.50063 0.92402,-0.36089 0.88005,0 0.0218,0.35024 0.19114,7.23842 0.26872,10.26553 0.0134,0.52414 0.42189,0.94671 0.94619,0.94671 0.52431,0 0.96031,-0.42257 0.94671,-0.94671 l -0.26871,-10.35648 -0.0698,-7.9251 2.09083,6.62389 c 0.11105,0.3518 0.55355,0.52963 0.90537,0.41858 l 0.0326,-0.0103 c 0.3518,-0.11105 0.54565,-0.48379 0.4346,-0.8356 l -2.71405,-8.59845 c -0.11661,-0.29672 -0.31809,-0.51893 -0.66714,-0.52193 h -1.79421 -1.10226 z m -1.80299,10.16217 v 8.35556 0.004 c 0.008,0.25214 0.1581,0.43067 0.2992,0.55087 0.14109,0.1202 0.28806,0.19703 0.42168,0.24236 l 0.19327,-0.56844 c -0.0509,-0.0173 -0.157,-0.0722 -0.22634,-0.13126 -0.068,-0.0579 -0.0851,-0.10488 -0.0858,-0.10903 v -8.34368 z m -1.36891,0.7152 v 3.92948 l 0.002,0.008 c 0.01,0.22857 0.12168,0.38622 0.24184,0.52348 0.12015,0.13725 0.25853,0.24822 0.40825,0.32401 l 0.27181,-0.53537 c -0.0555,-0.0281 -0.15722,-0.10596 -0.22686,-0.18551 -0.0696,-0.0795 -0.0962,-0.16527 -0.0956,-0.15193 v -3.91243 z\\"/>
  </g>
</svg>",
    "name": "iso7001/ISO-7001-PI-BP-003.svg",
  },
]
`;
